### PR TITLE
deps: update to `form-js@0.9.2`

### LINKED ISSUE #3123: Upgrade to `form-js@0.8.0`
### What should we do?

Update to the newly released form-js@0.8.0 version.

### Why should we do it?

To integrate the new dynamic data, taglist and checklist functionality into Camunda Modeler that was made available with this latest version, as these features will launch on our products during the next alpha.

### PR DESCRIPTION
**Pull request by Skaiir**
Closes #3123

### CODE REVIEW

**Review by pinussilvestrus — COMMENTED** [review]
I'm not super up-to-date with the current linting state, so I'm asking:

Shouldn't we lint for `valuesKey` in the properties panel as well?

image

File: client/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "@bpmn-io/align-to-origin": "^0.7.0",
     "@bpmn-io/dmn-migrate": "^0.4.3",
     "@bpmn-io/extract-process-variables": "^0.5.1",
-    "@bpmn-io/form-js": "^0.7.2",
+    "@bpmn-io/form-js": "^0.8.0",
```
Comment (pinussilvestrus) [inline client/package.json:13]: Showing the errors in the properties panel is not yet supported, right? Do we have an issue with adding this feature to the linter (similar to BPMN)?

Sorry for asking now, and not in the linter integration PR!

**Comment by Skaiir** [comment]
@pinussilvestrus Gonna write here again for visibility but the fact that the rule doesn't trigger in the first screenshot is actually a bug, fixed by this PR https://github.com/camunda/form-linting/pull/9. I'll release a patch and bump the linter in this PR as a fix.

File: client/package.json (hunk 1 above)
Comment (Skaiir) [reply client/package.json:13]: Mhh actually this is something I hadn't really considered. So we're talking about these if I understand correctly:

image

@philippfromme In your opinion, what kind of workload are we looking at in terms of adding these to the forms editor? From what I remember in `bpmn` this was a significant effort, no?

@christian-konrad Do you expect these to be in the form editor at this stage? Or are we happy to move on without them for now and would rather tackle these in the future?

**Comment by pinussilvestrus** [comment]
[v0.9.0](https://github.com/bpmn-io/form-js/commit/10ea539989ea857adc7511e5f1a80a99c7ec96b1) can be used now as well (although it would have no effect in this PR).

**Comment by Skaiir** [comment]
@pinussilvestrus I've only done 0.8.0 to keep the css change within the scope of this version upgrade. If you want I can also upgrade to 0.9.0 but I think we should separate the commits.

**Comment by pinussilvestrus** [comment]
> @pinussilvestrus I've only done 0.8.0 to keep the css change and the upgrade within the same commit. If you want I can also upgrade to 0.9.0 but it'll be in a different one :)

Since the changes are only playground related, I can also handle that separately 👍  I will have a look at this PR today 🤞

**Comment by Skaiir** [comment]
Let's hold for just a bit on this one, I am confirming the status for https://jira.camunda.com/browse/CAM-14670.

**Review by marstamm — COMMENTED** [review]
Headers that are wrongfully marked as sticky are back:

Recording 2022-09-27 at 17 07 55

Should we try to solve these before we merge it?

**Comment by pinussilvestrus** [comment]
This would need a `@bpmn-io/properties-panel` upgrade in form-js, as it is not a peer dependency, yet (on purpose).

**Comment by Skaiir** [comment]
@marstamm Oh lord not again 😢. Who was it who solved the issues for bpmn and dmn again? I feel like that was @pinussilvestrus right?

**Comment by marstamm** [comment]
> Should we publish a new version of form-js (0.9.2) and integrate it straight away with this PR?

Yes, we should fix this bug and release it as a patch release.

The sticky headers are part of [properties-panel 0.19.0](https://github.com/bpmn-io/properties-panel/blob/main/CHANGELOG.md#0190). Seems like form-js is still using [0.14.0](https://github.com/bpmn-io/form-js/blob/develop/package.json#L56).

**Comment by nikku** [comment]
@Skaiir Please look into this one with high priority. Let's get this integrated as soon as possible, end-to-end.

**Review by nikku — CHANGES_REQUESTED** [review]
Cf. https://github.com/camunda/camunda-modeler/pull/3133#issuecomment-1260483233

**Comment by Skaiir** [comment]
Everything is now through, hopefully we've got it this time :)

**Comment by marstamm** [comment]
I can still reproduce it. Looking at the package-lock:

```
    "node_modules/@bpmn-io/form-js": {
      "version": "0.9.2",
      "resolved": "https://registry.npmjs.org/@bpmn-io/form-js/-/form-js-0.9.2.tgz",
      "integrity": "sha512-ffZPQ5H5PITs3cGIlnIuaUD1+DVCjk00g9aKdR360moeTyEOSPLn1Lm6nkSdkZ5QRnJdIa3iSN3JmOJeClZAWA==",
      "dependencies": {
        "@bpmn-io/form-js-editor": "^0.9.1",
        "@bpmn-io/form-js-playground": "^0.9.1",
        "@bpmn-io/form-js-viewer": "^0.9.1"
      }
    },
```

https://github.com/camunda/camunda-modeler/pull/3133/files#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519R4304-R4313

seems like only the parent was upgraded but the editor itself is still on version 0.9.1?

**Comment by Skaiir** [comment]
@marstamm Yeah good spot, the publish script didn't publish everything:

https://github.com/bpmn-io/form-js/commit/9fb8cf4a9bffe209ad7f76df8a9d764e1f782803#diff-f285967735af435cfb962418c7e2059db93e7a9310b6f09f4950d12c36682425
https://github.com/bpmn-io/form-js/commit/6a7ad7ea0554bf5afa9e27c3c203a7e3cc99f54f

I am curious as to why though. Okay time for more investigations :trollface:

**Review by marstamm — APPROVED** [review]
Can't replicate anymore 🚀

**Comment by pinussilvestrus** [comment]
Hooray 🎉

**Pull request merged**